pkg/githubclient/client_repository.go

```diff
@@ -52,3 +53,175 @@ func (c *Client) ListRepositories(ctx context.Context, owner string) ([]Reposito
+	repository.Archived = repositorySettings.Archived
+
+	// Merge settings
+	repository.AllowMergeCommit = repositorySettings.AllowMergeCommit
+	repository.AllowSquashMerge = repositorySettings.AllowSquashMerge
+	repository.AllowRebaseMerge = repositorySettings.AllowRebaseMerge
+
+	// Pull Requests
+	repository.AllowUpdateBranch = repositorySettings.AllowUpdateBranch
+	repository.AllowAutoMerge = repositorySettings.AllowAutoMerge
+	repository.DeleteBranchOnMerge = repositorySettings.DeleteBranchOnMerge
+
+	// This is required since Github does not allow overrides for flags specified
+	// at organization level.
+	// Otherwise you will run into the following error:
+	// HTTP 422 This organization does not allow private repository forking
+	repository.AllowForking = nil
+
+	underlyingClient := c.getUnderlyingClient(ctx)
+	repository, _, err := underlyingClient.Repositories.Edit(ctx, *repository.Owner.Login, *repository.Name, repository)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("set repository %s/%s settings", owner, repo))
+
+	return repository, nil
+}
+
+func (c *Client) SetRepositoryPermissions(ctx context.Context, repository *github.Repository, permissions map[string]string) error {
+	org := *repository.Organization.Login
+	owner := *repository.Owner.Login
+	repo := *repository.Name
+
+	//data, err := json.MarshalIndent(permissions, "", "  ")
+	//if err != nil {
+	//	return microerror.Mask(err)
+	//}
+	//fmt.Printf("permissions\n%s\n", data)
+
+	underlyingClient := c.getUnderlyingClient(ctx)
+
+	for teamSlug, permission := range permissions {
+		c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("grant %s permission to %s on %s/%s repository", permission, teamSlug, owner, repo))
+
+		opt := &github.TeamAddTeamRepoOptions{Permission: permission}
+
+		_, err := underlyingClient.Teams.AddTeamRepoBySlug(ctx, org, teamSlug, owner, repo, opt)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("granted %s permission to %s on %s/%s repository", permission, teamSlug, owner, repo))
+	}
+
+	return nil
+}
+
+func (c *Client) SetRepositoryBranchProtection(ctx context.Context, repository *github.Repository) error {
+	owner := *repository.Owner.Login
+	repo := *repository.Name
+	default_branch := *repository.DefaultBranch
+
+	False := false
+
+	opts := &github.ProtectionRequest{
+		RequiredPullRequestReviews: &github.PullRequestReviewsEnforcementRequest{
+			RequiredApprovingReviewCount: 1,
+		},
+		AllowForcePushes: &False,
+		AllowDeletions:   &False,
+		EnforceAdmins:    true,
+	}
+
+	checks, err := c.getGithubChecks(ctx, repository, default_branch)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// We can only set RequiredStatusChecks when there is at least 1 check available.
+	// Otherwise we hit a HTTP 422 Invalid request.
+	if len(checks) > 0 {
```

Comment:
Shouldn't it just return at this point?

Reply:
Well no because there are other settings we still want to set for the branch protection.